sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala

```diff
@@ -117,7 +117,8 @@ class OptimizeMetadataOnlyQuerySuite extends QueryTest with SharedSparkSession {
 
   testNotMetadataOnly(
     "Don't optimize metadata only query for GroupingSet/Union operator",
-    "select partcol1, max(partcol2) from srcpart where partcol1 = 0 group by rollup (partcol1)",
+    "select partcol1, max(partcol2) from srcpart where partcol1 = 0 " +
+      "group by grouping sets (partcol1)",
```

Comment:
Do we need to change this?

Reply:
Yea, the original query was affected by this pr and the original test failed; this pr transformed `rollup` into `union(aggregate with keys, aggregate without keys)`, then the `aggregate without keys` side can be optimized by the `OptimizeMetadataOnlyQuery` rule, then `testNotMetadataOnly()` failed.

Reply:
Could you add that information into the PR description in order to make it as a commit log?

Reply:
sure!